alerting/src/test/kotlin/org/opensearch/alerting/AlertingRestTestCase.kt

```diff
@@ -598,7 +598,7 @@ abstract class AlertingRestTestCase : ODFERestTestCase() {
     protected fun indexDoc(index: String, id: String, doc: String, refresh: Boolean = true): Response {
         val requestBody = StringEntity(doc, APPLICATION_JSON)
         val params = if (refresh) mapOf("refresh" to "true") else mapOf()
-        val response = client().makeRequest("PUT", "$index/_doc/$id", params, requestBody)
+        val response = adminClient().makeRequest("PUT", "$index/_doc/$id", params, requestBody)
```

Comment:
Do you know if changing this to `adminClient()` has any side-effects to the other tests that were using `indexDoc` already? As in, they were trying to index against a regular index with some user in a security test and we expect it to only pass if they have permissions but now it'll always work with `adminClient()`?

If we want to keep this generic, we can overload the function to keep the old behavior and have another method for the `DestinationMigrationUtilServiceIT` cases:

```kotlin
protected fun indexDoc(index: String, id: String, doc: String, refresh: Boolean = true): Response {
    return indexDoc(client(), index, id, doc, refresh)
}

protected fun indexDocWithAdminClient(index: String, id: String, doc: String, refresh: Boolean = true): Response {
	return indexDoc(adminClient(), index, id, doc, refresh)
}

private fun indexDoc(client: RestClient, index: String, id: String, doc: String, refresh: Boolean): Response {
    val requestBody = StringEntity(doc, APPLICATION_JSON)
    val params = if (refresh) mapOf("refresh" to "true") else mapOf()
    val response = client().makeRequest("PUT", "$index/_doc/$id", params, requestBody)
    assertTrue(
        "Unable to index doc: '${doc.take(15)}...' to index: '$index'",
        listOf(RestStatus.OK, RestStatus.CREATED).contains(response.restStatus())
    )
    return response
}
```

Then you can use `indexDocWithAdminClient` in `DestinationMigrationUtilServiceIT`

Reply:
thanks Mohammad for pointing that out. changed it now.